Avoid poisoning the trusted certificate cache on error (#31438)
hashicorp/vault: changelog/31438.txt

```release-note:bug

> auth/cert: Do not poison the cache of trusted certificates if one or more certificates fails to load.

Reviewer: ```suggestion
auth/cert: Recover from partially populated caches of trusted certificates if one or more certificates fails to load.
```

We aren't injecting incorrect/bad entries within the cache which is what I'd associate cache poisoning
